packages/aws-cdk/lib/api/hotswap/stepfunctions-state-machines.ts

```diff
@@ -12,14 +12,18 @@ export async function isHotswappableStateMachineChange(
   }
 
   const machineNameInCfnTemplate = change.newValue?.Properties?.StateMachineName;
-  const machineName = await establishResourcePhysicalName(logicalId, machineNameInCfnTemplate, evaluateCfnTemplate);
-  if (!machineName) {
+
+  const machineArn = machineNameInCfnTemplate ? await evaluateCfnTemplate.evaluateCfnExpression({
+    'Fn::Sub': 'arn:${AWS::Partition}:states:${AWS::Region}:${AWS::AccountId}:stateMachine:' + machineNameInCfnTemplate,
+  }) : await establishResourcePhysicalName(logicalId, machineNameInCfnTemplate, evaluateCfnTemplate);
```

Comment:
Does it really make sense to call `establishResourcePhysicalName()` here (in the second branch of the ternary expression)? 

We know that `machineNameInCfnTemplate` is `undefined` in this branch, which means [this code](https://github.com/aws/aws-cdk/blob/8e3269b11ae013c74fd46046697ebd4d647bb7ed/packages/aws-cdk/lib/api/hotswap/common.ts#L66) will simply call `evaluateCfnTemplate.findPhysicalNameFor(logicalId);`.

Given that, can we just do that ourselves instead? Otherwise, this is pretty confusing code.